### PR TITLE
docs: add frontend UI and usability guide

### LINKED ISSUE #918: Frontend usability guideline
### Tasks

 - [x] Describe Usability, UI and UX terminology
 - [x] Write UI guideline
 - [x] Write usability guideline

### Use case

_No response_

### Related Pull request

#999

### PR DESCRIPTION
**Pull request by tschawax**
Closes #918

## Basics

- [x] I added a line to [changelog.md](/doc/changelog.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages and changelog.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] I've read through the whole documentation
- [ ] I've checked conformity to guidelines

### CODE REVIEW

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    logo <- 48px -> navigation-link-1 <- 24px -> navigation-link-2 <- 24px -> navigation-link-3
+
+- **Line height**  
+  The bigger the text the smaller the line height's value should be.
+
+  - Headings: 1.1 - 1.3
+  - Running text: 1.3 - 1.5
+
+- **Letter spacing**  
+  Often times, like in PermaplanT's current case, the default value (in Tailwind: _tracking-normal_) is good enough.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:49]: Maybe use code as then you still have emphasize available for other uses:

```suggestion
  Often times, like in PermaplanT's current case, the default value (in Tailwind: `tracking-normal`) is good enough.
```

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+**Only those** three colour palettes **should be used** throughout the whole application. The ratio of their current usage in PermaplanT is conforming nicely to the popular **60-30-10** design rule.
+
+### Forms
+
+#### Design
+
+- Display a **centered heading** if the form is representing a whole page, e.g. PermaplanT's forms to create maps and seeds.
+- Add a **capitalized label** above each field.
+- **Align** labels to the **left** of their corresponding fields.
+- Always explicitly **associate** the **label with** the belonging **element** by using the label's _for_ attribute (in React: _htmlFor_).
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:70]: Sentence complicated and bold formatting seems to me overused here.

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  This enables focusing the input field by clicking/tapping on the label.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:71]: Maybe add something like this (everywhere where rationale is given) to make the guideline easier to read:

```suggestion
  _Rationale:_ This enables focusing the input field by clicking/tapping on the label.
```

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  It also adds screen reader support.
+- All input fields should have the **same styling**, i.e. border, colour, text size, padding, etc.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:73]: Yes, please elaborate which styling.

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+- Use **opacity** for an input field's **placeholder** text.
+
+#### Error Prevention
+
+- Use the **corresponding _type_** of the input field, e.g. _date_ for date fields, _number_ for numerical-only inputs etc.
+- Mark required fields with an **asterisk**.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:79]: Color etc..

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+- Use HTML-native **constraint** attributes for input fields where appropriate, e.g. _maxlength_, _required_, _step_, etc.
+- Use short **placeholder** texts for input fields to show allowed values or further explain the intent of the input field.
+- Users should **never** be **required** to manually enter a **metrical** unit or a **currency** symbol.
+  Instead, make the unit part of the input field itself or leave it out if it's clear from the context.
+- **Submission** to the backend should not happen until all fields are **verified** by the **frontend** logic.
+- **Double submission** should be **prevented**, e.g. by disabling the submission button until the server has responded.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:85]: Are there other good options?

```suggestion
- **Double submission** should be **prevented** by disabling the submission button until the server has responded.
```

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+### Consistency
+
+Things with the **same meaning** should **look and behave** the **same** throughout the whole application.
+Users should never be uncertain if different words mean the same thing or different actions trigger the same process.
+
+Examples:
+
+- Call-to-action buttons should look the same everywhere.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:94]: How? (For all these examples)

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+- The layout of forms and the styling of their input fields should be the same in every form.
+- Validation errors in forms should be rendered the same way in any form.
+- Toast errors should always be shown in the same location.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:97]:
```suggestion
- Toast errors should always be shown on the top right.
```

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+- Headings of different pages should be positioned in the same location and have the same styling.
+- The same application-specific terms should be used everywhere to communicate to the user.
+
+### Icons
+
+- In their **passive**, i.e. currently not active/enabled, state they should appear in **neutral** colours/colourless.
+- When **active**, i.e. currently activated/enabled, they should get a small **highlighting** through a visually stronger and more colourful appearance by using the design's primary/secondary colours.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:104]: when primary when secondary?

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+- When **disabled**, i.e. currently not clickable, they should be **greyed-out**, either via decreasing its opacity or choosing light, faint colours.  
+  The mouse **cursor** should be styled with the **_not-allowed_** css property.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:106]:
```suggestion
  The mouse cursor should be styled with the `not-allowed` CSS property.
```

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+- A **tooltip** on hovering should be displayed for every icon in PermaplanT's toolbox.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:107]: Can you please define must/should(as needed) for all the guidelines.

```suggestion
- A **tooltip** on hovering must be displayed for every icon in PermaplanT's toolbox.
```

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+
+Highlighting techniques:
+
+- Font weight:
+
+  - **Bold** text adds tolerable noise to the design and enables distinct highlighting in both short and long texts.
+    In general, it is the preferred highlighting technique to use.
+  - _Italic_ text adds minimal noise to the design, but lacks in recognizability.
+  - <u>Underlining</u> adds most noise and compromises a text's legibility.
+    It should hardly ever be used.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:120]:
```suggestion
    It shouldn't be used.
```

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+- **concise**: the more text the less likely it will be read by the user
+- **clear**: straight to the message's essence
+- **understandable** for technical laymen: no error codes or technical terms
+- **no exclamation** marks: interpreted as commanding
+- **no uppercase** words: comes across as shouting
+- **detailed** information should be hidden behind a **read more** link or a **collapsed** section
+- **Headlines**, **sub-headlines** and **labels** should be **capitalized**
+- **Minimize hyphenation** of words
+
+### Error Messages
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:141]: Overlap to the other UI document, can you move it here? Do we really want everything in one document?

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+In general, error messages often convey the impression that an application's stability is weak which, in consequence, leads to a **declined user experience**.
+Error messages should **only** be used **if** the **user**, without that information, **would be badly surprised** by the result of an action.
+
+Error messages should fulfill following criteria:
+
+- **polite** and **neutral**: do not directly or indirectly blame the user and stay away from jokes
+- **brief**
+- **specific** to the problem
+- contain **reason for the problem**
+- **solution-oriented**: ideally, a solution to the occurred problem is offered
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:152]: This is dangerous: Solutions should be only offered if you are sure about the solution.

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+- no **privacy-violating** information: e.g. informing the user upon a failed login that the entered email does not exist in the system - this could additionally also encourage people to scan our database for emails
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:153]: As keycloak (hopefully) already does this for us, I don't think we need this guideline.

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+
+### Search Fields
+
+To prevent _Chromium_ and _WebKit_ based browsers from adding their own cross icon and functionality into search input fields, **do not declare** input fields with **type _search_**.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:157]: Start with important+general information, rationale+details later.

```suggestion
Do not declare input fields with type `search`.
_Rationale:_  To prevent _Chromium_ and _WebKit_ based browsers from adding their own cross icon and functionality into search input fields.
```

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+- **brief**
+- **specific** to the problem
+- contain **reason for the problem**
+- **solution-oriented**: ideally, a solution to the occurred problem is offered
+- no **privacy-violating** information: e.g. informing the user upon a failed login that the entered email does not exist in the system - this could additionally also encourage people to scan our database for emails
+
+### Search Fields
+
+To prevent _Chromium_ and _WebKit_ based browsers from adding their own cross icon and functionality into search input fields, **do not declare** input fields with **type _search_**.  
+Instead, **use** PermaplanT's **SearchInput** component which renders its own cross icon, together with a search icon and the input field itself.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:158]: Why is "use" bold?

```suggestion
Instead, we use PermaplanT's **SearchInput** component which renders its own cross icon, together with a search icon and the input field itself.  
```

File: doc/guidelines/frontend-ui-usability-guide.md [hunk 18]
```diff
@@ -0,0 +1,166 @@
...
+To prevent _Chromium_ and _WebKit_ based browsers from adding their own cross icon and functionality into search input fields, **do not declare** input fields with **type _search_**.  
+Instead, **use** PermaplanT's **SearchInput** component which renders its own cross icon, together with a search icon and the input field itself.  
+That way we provide the same search input functionality for every browser.
+
+The **SearchInput** component has following **behavior** and **functionality**:
+
+- The cross icon will be shown if there is any search term entered into the input field, and hidden otherwise.
+- Clicking on the cross icon will reset the current search term and keep the focus on the input field, so the user can continue typing.
+- Pressing the _ESC_ key likewise resets the current search term and keeps the focus on the input field.
+- The _SearchInput_ component exposes a method which enables parent components to set the focus on it, besides that it is fully encapsulated, i.e. it hides all its DOM nodes from the outside.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability-guide.md:166]: This should probably better be in the components description, shouldn't it?

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great work, very good start! :heart: 

- Some points could be more specific and hands-on. (Simply say specific values which are already successfully used and look nice.)
- I think you can rewrite doc/guidelines/ui.md, it had more or less the same goal as this document. Maybe we also don't need doc/guidelines/frontend.md anymore?
- We also need to be careful about overlap to other documents outside of guidelines, in particular component documentation (storybook). If guidelines are already fulfilled by using one of our components, you basically only need to say "use that component". The guidelines should be as short as possible, as it needs to be manually reviewed.

File: doc/guidelines/frontend-ui-usability.md [hunk 19]
```diff
@@ -0,0 +1,177 @@
...
+The following guidelines are:
+
+- on the one hand a mix of general UI and usability best practices and
+- on the other hand design suggestions tailored to PermaplanT.
+
+### Text
+
+- **Alignment**
+
+  - Left align when text spans over more than three lines because centered text becomes hard to follow when having to jump to the next line multiple times.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability.md:29]: Formulate everything as rule if possible. So is here the rule that we should be below three lines if possible? Which alignment to choose if we are above 3 lines?

File: doc/guidelines/frontend-ui-usability.md [hunk 20]
```diff
@@ -0,0 +1,177 @@
...
+  - Avoid mixing two different alignments in a text section, e.g. a centered headline should be followed by centered text.
+
+- **Spacing / margin**  
+  Multiplier strategy to calculate the margins - if in doubt, use a multiplier of 2, e.g.:
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability.md:33]: Can we give concrete guidelines which px to use?

```suggestion
  Multiplier strategy to calculate the margins - if in doubt, use a multiplier of 2. Generally we use:
```

File: doc/guidelines/frontend-ui-usability.md [hunk 21]
```diff
@@ -0,0 +1,177 @@
...
+
+  - Text sections:  
+    headline-1 <- 16px -> text <- 32px -> headline-2
+  - Navigation bar:  
+    logo <- 48px -> navigation-link-1 <- 24px -> navigation-link-2 <- 24px -> navigation-link-3
+
+- **Letter spacing**  
+  Often times, like in PermaplanT's current case, the default value (in Tailwind: `tracking-normal`) is good enough.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability.md:41]: Is the list below exceptions?

```suggestion
  Often times, like in PermaplanT's current case, the default value (in Tailwind: `tracking-normal`) is good enough.
  Exceptions:
```

File: doc/guidelines/frontend-ui-usability.md [hunk 22]
```diff
@@ -0,0 +1,177 @@
...
+  - Headings: negative values can be used for the bigger headings h1-h3, i.e. `tracking-tighter` or `tracking-tight`
+  - Running text: usually can be left as is but, if needed, can be set to `tracking-wide` or `tracking-wider`
+
+### Colors
+
+PermaplanT uses following **three color palettes**:
+
+- **Neutral** ~60% (PermaplanT: gray): **dominant** color of the design, e.g. used for background, text and labels
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability.md:49]: What means the 60%? How much it is used?

File: doc/guidelines/frontend-ui-usability.md [hunk 23]
```diff
@@ -0,0 +1,177 @@
...
+  To achieve this, we are simply using the default setting of our used library (_react-toastify_).
+- Use the same **application-specific terms** everywhere when communicating to the user, e.g. always write _PermaplanT_.
+- **Forms** and their fields should look and behave the same everywhere according to our guidelines in [Forms](#forms).
+- In every **form**, **validation** errors are detected and shown by using the native input validation of HTML5.
+
+### Icons
+
+- In their **passive**, i.e. currently not active/enabled, state they must appear in **neutral** colors/colorless.
+- When **active**, i.e. currently activated/enabled, they must get a small **highlighting** through a visually stronger and more colorful appearance by using the design's primary colors.
+- When **disabled**, i.e. currently not clickable, they must be **greyed-out**, either via decreasing its opacity or choosing light, faint colors.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability.md:120]: How much decreasing? Why having two options?

File: doc/guidelines/frontend-ui-usability.md [hunk 24]
```diff
@@ -0,0 +1,177 @@
...
+- In every **form**, **validation** errors are detected and shown by using the native input validation of HTML5.
+
+### Icons
+
+- In their **passive**, i.e. currently not active/enabled, state they must appear in **neutral** colors/colorless.
+- When **active**, i.e. currently activated/enabled, they must get a small **highlighting** through a visually stronger and more colorful appearance by using the design's primary colors.
+- When **disabled**, i.e. currently not clickable, they must be **greyed-out**, either via decreasing its opacity or choosing light, faint colors.  
+  The mouse cursor must be styled with the `not-allowed` css property.
+- A **tooltip** on hovering must be displayed for every icon in PermaplanT's toolbox.
+  The tooltip contains the icon's **label** (as concise as possible) and, if existing, the assigned **shortcut**.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability.md:123]: Preferably we should order the guidelines by priority. E.g. isn't this more important then the colors for passive/active/disabled?

File: doc/guidelines/frontend-ui-usability.md [hunk 25]
```diff
@@ -0,0 +1,177 @@
...
+    In general, it is the preferred highlighting technique to use.
+  - _Italic_ text adds minimal noise to the design, but lacks in recognizability.
+  - <u>Underlining</u> adds most noise and compromises a text's legibility.
+    It should not be used.
+
+- Colors: primary and secondary colors of the design's color palette
+
+- Images: users generally remember images better than words
+
+- Avoid different fonts for highlighting purposes.
```
Comment (markus2330) [inline doc/guidelines/frontend-ui-usability.md:141]: Are different fonts allowed for any purpose?

**Review by markus2330 — CHANGES_REQUESTED** [review]
The guidelines are getting great! :rocket: Please rebase (there are many unwanted changes listed in this PR). In the next round you can also let review others. All the frontend devs should review it.

**Comment by markus2330** [comment]
Maybe rebase/updating branch is enough. If not, yes please create new PR.

**Comment by markus2330** [comment]
jenkins build please

File: doc/guidelines/frontend-ui-usability.md [hunk 26]
```diff
@@ -0,0 +1,195 @@
...
+    - Guided tour: center highlighted actions
+
+- **Spacing / margin**
+
+  - `<h1>` Headings: `mb-8`
+  - `<h2>` Subheadings: `mb-2`
+  - Cards (like on our Map overview): `mb-4`
+  - Form fields (like on the Seeds page): `mb-8`
+  - Labels of form fields: `mb-2`
+  - If in doubt, use the multiplier strategy to calculate margins between sections, e.g. we are using a multiplier of 4 in our navigation bar:
```
Comment (absurd-turtle) [inline doc/guidelines/frontend-ui-usability.md:50]: It is unclear to me what the multiplier strategy is.

File: doc/guidelines/frontend-ui-usability.md [hunk 27]
```diff
@@ -0,0 +1,195 @@
...
+  - Cards (like on our Map overview): `mb-4`
+  - Form fields (like on the Seeds page): `mb-8`
+  - Labels of form fields: `mb-2`
+  - If in doubt, use the multiplier strategy to calculate margins between sections, e.g. we are using a multiplier of 4 in our navigation bar:  
+    logo <- 48px -> navigation-link-1 <- 24px -> navigation-link-2 <- 24px -> navigation-link-3
+
+- **Letter spacing**
+
+  - Always go with the default value, i.e. `tracking-normal`.
+  - In case letter spacing changes are considered in the future:
```
Comment (absurd-turtle) [inline doc/guidelines/frontend-ui-usability.md:56]: I don't think we need considerations for the future in this document.
It it is more effort to read through the guideline and it doesn't apply anyways.

Comment (absurd-turtle) [reply doc/guidelines/frontend-ui-usability.md:56]: I agree with the statement though :D

File: doc/guidelines/frontend-ui-usability.md [hunk 28]
```diff
@@ -0,0 +1,195 @@
...
+- **Neutral** (PermaplanT: gray): **dominant** color of the design, e.g. used for background, text and labels.  
+  Optimal usage: ~60%
+- **Primary** (PermaplanT: asparagus green): PermaplanT's **main brand** color, e.g. used for buttons, hovering, highlighting focused input fields.  
+  Optimal usage: ~30%
+- **Secondary** (PermaplanT: sea blue): PermaplanT's **second brand** color, e.g. used for highlighting a few UI elements (e.g. in the layers section) and action-texts.  
+  Optimal usage: ~10%
+
+Check out Google's [Material Design Guidelines](https://m3.material.io/styles/color/the-color-system/key-colors-tones) for more information on color palettes and their usage.
+
+**Only those** three color palettes **should be used** throughout the whole application.
```
Comment (absurd-turtle) [inline doc/guidelines/frontend-ui-usability.md:73]: I disagree with this statement. It occasionally makes sense to use other colors which have some sort of association. E.g. usage of the color red to indicate errors or destructive actions.

File: doc/guidelines/frontend-ui-usability.md [hunk 29]
```diff
@@ -0,0 +1,195 @@
...
+### Colors
+
+PermaplanT uses following **three color palettes**:
+
+- **Neutral** (PermaplanT: gray): **dominant** color of the design, e.g. used for background, text and labels.  
+  Optimal usage: ~60%
+- **Primary** (PermaplanT: asparagus green): PermaplanT's **main brand** color, e.g. used for buttons, hovering, highlighting focused input fields.  
+  Optimal usage: ~30%
+- **Secondary** (PermaplanT: sea blue): PermaplanT's **second brand** color, e.g. used for highlighting a few UI elements (e.g. in the layers section) and action-texts.  
+  Optimal usage: ~10%
```
Comment (absurd-turtle) [inline doc/guidelines/frontend-ui-usability.md:69]: Please add some examples on when to use the main brand vs the second brand color.

File: doc/guidelines/frontend-ui-usability.md [hunk 30]
```diff
@@ -0,0 +1,195 @@
...
+    - Running text: `tracking-wide` or `tracking-wider`
+
+### Colors
+
+PermaplanT uses following **three color palettes**:
+
+- **Neutral** (PermaplanT: gray): **dominant** color of the design, e.g. used for background, text and labels.  
+  Optimal usage: ~60%
+- **Primary** (PermaplanT: asparagus green): PermaplanT's **main brand** color, e.g. used for buttons, hovering, highlighting focused input fields.  
+  Optimal usage: ~30%
```
Comment (absurd-turtle) [inline doc/guidelines/frontend-ui-usability.md:67]: I don't agree with the percentages. I don't think they are helpful.

According to the material guideline the primary color should be used for the most prominent/important actions and therefore might be used less than the secondary color.

File: doc/guidelines/frontend-ui-usability.md [hunk 31]
```diff
@@ -0,0 +1,195 @@
...
+  To achieve this, we are simply using the default setting of our used library (_react-toastify_).
+- Use the same **application-specific terms** everywhere when communicating to the user, e.g. always write _PermaplanT_.
+- **Forms** and their fields should look and behave the same everywhere according to our guidelines in [Forms](#forms).
+- In every **form**, **validation** errors are detected and shown by using the native input validation of HTML5.
+
+### Icons
+
+- A **tooltip** on hovering must be displayed for every icon in PermaplanT's toolbox.
+  The tooltip contains the icon's **label** (as concise as possible) and, if existing, the assigned **shortcut**.
+- In their **passive**, i.e. currently not active/enabled, state they must appear in **neutral** colors/colorless.
```
Comment (absurd-turtle) [inline doc/guidelines/frontend-ui-usability.md:138]:
```suggestion
- In their **passive**, i.e. currently not active/enabled state they must appear in **neutral** colors/colorless.
```

File: doc/guidelines/frontend-ui-usability.md [hunk 32]
```diff
@@ -0,0 +1,195 @@
...
+    In general, it is the preferred highlighting technique to use.
+  - _Italic_ text adds minimal noise to the design, but lacks in recognizability.
+  - <u>Underlining</u> adds most noise and compromises a text's legibility.
+    It should not be used.
+
+- Colors: primary and secondary colors of the design's color palette
+
+- Images: users generally remember images better than words
+
+- Never use a different font family.
```
Comment (absurd-turtle) [inline doc/guidelines/frontend-ui-usability.md:159]: Which font-family is / should be used?
Is there one defined? Do we want one or fallback to the system font of the user?
Do we want to have a different font-family for headlines?

File: doc/guidelines/frontend-ui-usability.md [hunk 33]
```diff
@@ -0,0 +1,195 @@
...
+
+- **concise**: the more text the less likely it will be read by the user
+- **clear**: straight to the message's essence
+- **understandable** for technical laymen: no status codes or technical terms
+- **no exclamation** marks: interpreted as commanding
+- **no uppercase** words: comes across as shouting
+- **detailed** information should be hidden behind a **read more** link or a **collapsed** section
+- **Headlines**, **sub-headlines** and **labels** should be **capitalized**
+- **Minimize hyphenation** of words
+- **American English** spelling
```
Comment (absurd-turtle) [inline doc/guidelines/frontend-ui-usability.md:173]: and german

**Review by absurd-turtle — CHANGES_REQUESTED** [review]
Nice work :)
Most of the guidelines make sense to me and the document is not too long.
We should try to implement as much as we can in the basic components and add in the document when the guidelines are fulfilled by default.
So we don't have to worry about it too much when creating new features.

File: doc/guidelines/frontend-ui-usability.md (hunk 27 above)
Comment (markus2330) [reply doc/guidelines/frontend-ui-usability.md:56]: We could move future guidelines to a final section?

File: doc/guidelines/frontend-ui-usability.md (hunk 28 above)
Comment (markus2330) [reply doc/guidelines/frontend-ui-usability.md:73]: Then we probably should define more colors, e.g. the colors used for the companions? But I agree with @tschawax  that we shouldn't leave it to the respective devs to bring arbitrary other colors into the app as wanted. If anywhere the color for e.g. the companions is needed again, it should be the same color as already used.

File: doc/guidelines/frontend-ui-usability.md (hunk 33 above)
Comment (markus2330) [reply doc/guidelines/frontend-ui-usability.md:173]:
```suggestion
- **American English** and **German** (Neue Rechtschreibung) spelling
```

File: doc/guidelines/frontend-ui-usability.md [hunk 34]
```diff
@@ -0,0 +1,195 @@
...
+  - Labels of form fields: `mb-2`
+
+- **Letter spacing**
+
+  - Always go with the default value, i.e. `tracking-normal`.
+  - In case letter spacing changes are considered in the future:
+    - Headings: use negative values for the bigger headings h1-h3, i.e. `tracking-tighter` or `tracking-tight`
+    - Running text: `tracking-wide` or `tracking-wider`
+
+### Colors
```
Comment (4ydan) [inline doc/guidelines/frontend-ui-usability.md:63]: Which color should be used for cancel buttons e.g (cancel tour, cancel map/seed creation)?

File: doc/guidelines/frontend-ui-usability.md [hunk 35]
```diff
@@ -0,0 +1,195 @@
...
+
+- Font weight:
+
+  - **Bold** text adds tolerable noise to the design and enables distinct highlighting in both short and long texts.
+    In general, it is the preferred highlighting technique to use.
+  - _Italic_ text adds minimal noise to the design, but lacks in recognizability.
+  - <u>Underlining</u> adds most noise and compromises a text's legibility.
+    It should not be used.
+
+- Colors: primary and secondary colors of the design's color palette
```
Comment (4ydan) [inline doc/guidelines/frontend-ui-usability.md:156]: I personally think blue is not the best color for text highlighting, because of the resemblance to hyperlinks.

File: doc/guidelines/frontend-ui-usability.md [hunk 36]
```diff
@@ -0,0 +1,195 @@
...
+- Use short **placeholder** texts for input fields to show allowed values or further explain the intent of the input field.
+- Users should **never** be **required** to manually enter a **metrical** unit or a **currency** symbol.
+  Instead, make the unit part of the input field itself or leave it out if it's clear from the context.
+- **Submission** to the backend should not happen until all fields are **verified** by the **frontend** logic.
+- In case of submissions **on-the-fly**, e.g. setting dates for plants, the data should always be submitted with **debouncing**.
+- **Double submission** should be **prevented** by disabling the submission button until the server has responded.
+
+### Consistency
+
+Things with the **same meaning** should **look and behave** the **same** throughout the whole application.
```
Comment (4ydan) [inline doc/guidelines/frontend-ui-usability.md:119]: Which "things"? Words and actions?

File: doc/guidelines/frontend-ui-usability.md [hunk 37]
```diff
@@ -0,0 +1,195 @@
...
+- Use the **corresponding _type_** of the input field, e.g. _date_ for date fields, _number_ for numerical-only inputs etc.
+- Mark required fields with an **asterisk**:
+  - color: `text-red-500`
+  - all other styles are inherited from the parent label
+- Use HTML-native **constraint** attributes for input fields where appropriate, e.g. _maxlength_, _required_, _step_, etc.
+- Use short **placeholder** texts for input fields to show allowed values or further explain the intent of the input field.
+- Users should **never** be **required** to manually enter a **metrical** unit or a **currency** symbol.
+  Instead, make the unit part of the input field itself or leave it out if it's clear from the context.
+- **Submission** to the backend should not happen until all fields are **verified** by the **frontend** logic.
+- In case of submissions **on-the-fly**, e.g. setting dates for plants, the data should always be submitted with **debouncing**.
```
Comment (4ydan) [inline doc/guidelines/frontend-ui-usability.md:114]: Debouncing should be a general idea (e.g for search bars), not only for forms.

File: doc/guidelines/frontend-ui-usability.md [hunk 38]
```diff
@@ -0,0 +1,195 @@
...
+  - border focus: `focus:border-primary-500 dark:focus:border-primary-300`
+  - border radius: `rounded-lg`
+  - padding: `p-2.5`
+  - placeholder: `placeholder-neutral-300`
+  - text: `text-sm`
+
+#### Error Prevention
+
+- Use the **corresponding _type_** of the input field, e.g. _date_ for date fields, _number_ for numerical-only inputs etc.
+- Mark required fields with an **asterisk**:
```
Comment (4ydan) [inline doc/guidelines/frontend-ui-usability.md:106]: The asterisk color is already mentioned in colors.

File: doc/guidelines/frontend-ui-usability.md [hunk 39]
```diff
@@ -0,0 +1,195 @@
...
+    - Running text: `tracking-wide` or `tracking-wider`
+
+### Colors
+
+PermaplanT uses following **color palettes**:
+
+- **Neutral** (PermaplanT: gray): **dominant** color of the design, e.g. used for background, text and labels.
+- **Primary** (PermaplanT: asparagus green): PermaplanT's **main brand** color, e.g. used for call-to-action buttons, checkmarks, highlighting focused form input fields, hovering over icons, highlighting active/enabled icons in the toolbox.
+- **Secondary** (PermaplanT: sea blue): PermaplanT's **second brand** color, e.g. used for highlighting selected plants on the map, currently focused icons, spinners, action-texts in the guided tour and bigger UI elements like the transparency bars to have some contrast to the primary color.
+- Tailwind's default color palette for **red** is e.g. used to render validation errors in forms, form field asterisks, buttons for destructive actions.
```
Comment (danielsteinkogler) [inline doc/guidelines/frontend-ui-usability.md:70]: maybe we could mention the exact color that should be user for the different use cases. For example you mentioned below that the astrisks should be red-500, but i would excpect this here in the color section.

File: doc/guidelines/frontend-ui-usability.md (hunk 24 above)
Comment (4ydan) [reply doc/guidelines/frontend-ui-usability.md:138]: I agree. Maybe also with a table of contents, so someone can quickly find what they are searching.

**Review by 4ydan — CHANGES_REQUESTED** [review]
Great work 👍🏻, some small ideas from my side.

File: doc/guidelines/frontend-ui-usability.md [hunk 40]
```diff
@@ -0,0 +1,195 @@
+# Frontend UI & Usability
+
+## Terminology
```
Comment (4ydan) [inline doc/guidelines/frontend-ui-usability.md:3]: Maybe the terminology is better suited inside our glossary.
